source/common/ur_util.hpp

```diff
@@ -87,23 +91,120 @@ inline std::string create_library_path(const char *name, const char *path) {
 #endif
 
 ///////////////////////////////////////////////////////////////////////////////
-inline bool getenv_tobool(const char *name) {
-    const char *env = nullptr;
-
+inline std::optional<std::string> ur_getenv(const char *name) {
 #if defined(_WIN32)
-    char buffer[8];
-    auto rc = GetEnvironmentVariable(name, buffer, 8);
-    if (0 != rc && rc <= 8) {
-        env = buffer;
+    constexpr int buffer_size = 1024;
+    char buffer[buffer_size];
+    auto rc = GetEnvironmentVariable(name, buffer, buffer_size);
+    if (0 != rc && rc < buffer_size) {
+        return std::string(buffer);
+    } else if (rc >= buffer_size) {
+        std::cerr << "Environment variable " << name << " value too long!"
```

Comment:
please change it to an exception, so we don't print to screen. same below.

Ultimately, I think we shouldn't be as liberal with throwing exceptions as in this PR since ultimately the library interacts through a C API, but for now let's leave it as is. I wish we had `std::expected`... we might think about refactoring this with something simple based on `std::variant`.